src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EmitterExtensions.cs

```diff
@@ -30,4 +31,13 @@ public static bool IsSerializableJsonResponse(this EndpointResponse endpointResp
         }
         return false;
     }
+
+    public static string EmitHandlerArgument(this EndpointParameter endpointParameter) => $"{endpointParameter.SymbolName}_local";
+
+    public static string EmitArgument(this EndpointParameter endpointParameter) => endpointParameter.Source switch
+    {
+        EndpointParameterSource.JsonBody or EndpointParameterSource.Route or EndpointParameterSource.RouteOrQuery or EndpointParameterSource.JsonBodyOrService or EndpointParameterSource.FormBody => endpointParameter.IsOptional ? endpointParameter.EmitHandlerArgument() : $"{endpointParameter.EmitHandlerArgument()}!",
+        EndpointParameterSource.Unknown => throw new Exception("Unreachable!"),
```

Comment:
Oh I forgot the generator targets ns2.0

Reply:
Yep, we have to resort to this in a few places in the code. Considering the wider effort to remove `Exception` from the repo, we may want to come back here and create our own definition of the type to use within the generator code.

Reply:
```suggestion
        EndpointParameterSource.Unknown => throw new NotImplementedException("Unreachable!"),
```

Let's get ahead of it rather than come back to it. I don't want this to get merged at the same time as #47873 and start causing build failures. Chances are that it would conflict anyway, but just in case.